libraries/botbuilder/src/skills/skillHandlerImpl.ts

```diff
@@ -0,0 +1,212 @@
+    }
+
+    async onDeleteActivity(claimsIdentity: ClaimsIdentity, conversationId: string, activityId: string): Promise<void> {
+        return this.continueConversation(claimsIdentity, conversationId, (context) =>
+            context.deleteActivity(activityId)
+        );
+    }
+
+    private async getSkillConversationReference(conversationId: string): Promise<SkillConversationReference> {
+        let skillConversationReference: SkillConversationReference;
+
+        try {
+            skillConversationReference = await this.conversationIdFactory.getSkillConversationReference(conversationId);
+        } catch (err) {
+            // If the factory has overridden getSkillConversationReference, call the deprecated getConversationReference().
+            // In this scenario, the oAuthScope paired with the ConversationReference can only be used for talking with
+            // an official channel, not another bot.
+            if (err.message === 'NotImplemented') {
+                skillConversationReference = {
+                    conversationReference: await this.conversationIdFactory.getConversationReference(conversationId),
+                    oAuthScope: this.getOauthScope(),
+                };
+            } else {
+                throw err;
+            }
+        }
+
+        if (!skillConversationReference) {
+            throw new Error('skillConversationReference not found');
+        } else if (!skillConversationReference.conversationReference) {
+            throw new Error('conversationReference not found.');
+        }
+
+        return skillConversationReference;
+    }
+
+    private async processActivity(
+        claimsIdentity: ClaimsIdentity,
+        conversationId: string,
+        replyToActivityId: string,
+        activity: Activity
+    ): Promise<ResourceResponse> {
+        let resourceResponse: ResourceResponse;
+
+        await this.continueConversation(claimsIdentity, conversationId, async (context, ref) => {
+            /**
+             * This callback does the following:
+             *  - Applies the correct ConversationReference to the Activity for sending to the user-router conversation.
+             *  - For EndOfConversation Activities received from the Skill, removes the ConversationReference from the
+             *    ConversationIdFactory
+             */
+
+            const newActivity = TurnContext.applyConversationReference(activity, ref.conversationReference);
+
+            context.activity.id = replyToActivityId;
+            context.activity.callerId = `${CallerIdConstants.BotToBotPrefix}${JwtTokenValidation.getAppIdFromClaims(
+                claimsIdentity.claims
+            )}`;
+
+            // Legacy BotFrameworkAdapter flows rely on connector client being set inside turn context
+            if (this.adapter instanceof BotFrameworkAdapter) {
+                const client = this.adapter.createConnectorClient(newActivity.serviceUrl);
```

Comment:
Maybe try to push this into bot framework adapter itself, see https://github.com/microsoft/botbuilder-dotnet/blob/449a89e71dde3b2967a6cda8456f5fe4c2693df4/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L355

Reply:
Seems to exist in [`continueConversation` already](https://github.com/microsoft/botbuilder-js/blob/jpg/cloud-adapter/libraries/botbuilder/src/botFrameworkAdapter.ts#L415) so I'll remove this block.